gst/nnstreamer/tensor_query/nnstreamer_edge_common.c

```diff
@@ -217,3 +244,98 @@ nns_edge_data_add (nns_edge_data_h data_h, void *data, size_t data_len,
 
   return NNS_EDGE_ERROR_NONE;
 }
+
+/**
+ * @brief Get the nnstreamer edge data.
+ */
+int
+nns_edge_data_get (nns_edge_data_h data_h, void **data, size_t *data_len,
+    unsigned int nth)
+{
+  nns_edge_data_s *ed;
+
+  ed = (nns_edge_data_s *) data_h;
+
+  if (!NNS_EDGE_MAGIC_IS_VALID (ed)) {
+    nns_edge_loge ("Invalid param, given edge data is invalid.");
+    return NNS_EDGE_ERROR_INVALID_PARAMETER;
+  }
+
+  if (ed->num >= NNS_EDGE_DATA_LIMIT) {
```

Comment:
Is not "ed->num <= 0 " necessary for memory check as line 295 ?

Reply:
Thanks! Updated condition and error message.